src/arviz_plots/plots/khat_plot.py

```diff
@@ -0,0 +1,557 @@
+
+        if markersize is not None and "size" not in khat_aes:
+            khat_kwargs.setdefault("size", markersize)
+
+        marker_value = khat_kwargs.get("marker")
+
+        if backend == "matplotlib" and "marker" not in khat_aes:
+            marker_value = khat_kwargs.setdefault("marker", "+")
+
+        fillable_marker = marker_value is None or marker_value in _FILLABLE_MARKERS
+        default_color = khat_kwargs.get("color", color)
+
+        if default_color is None and "color" not in khat_aes:
+            default_color = _DEFAULT_POINT_COLOR
+
+        if backend == "matplotlib" and "color" not in khat_aes:
+            base_color = khat_kwargs.pop("color", default_color)
+            point_rgba = alpha_scaled_colors(base_color, khat_data.values, good_k_threshold)
+
+            if fillable_marker:
+                khat_kwargs.setdefault("facecolor", point_rgba)
+                khat_kwargs.setdefault("edgecolor", point_rgba)
+            else:
+                khat_kwargs.setdefault("color", point_rgba)
+        elif "color" not in khat_aes and default_color is not None:
+            khat_kwargs.setdefault("color", default_color)
+
+        plot_collection.map(
+            scatter_xy,
+            "khat",
+            data=khat_dataset,
+            ignore_aes=khat_ignore,
+            **khat_kwargs,
+        )
+
+    if show_hlines and hline_values:
+        hlines_kwargs = get_visual_kwargs(visuals, "hlines")
+
+        if hlines_kwargs is not False:
+            _, hlines_aes, _ = filter_aes(plot_collection, aes_by_visuals, "hlines", [])
+            linestyle_cycle = [":", "-.", "--", "-"]
+
+            for idx, value in enumerate(hline_values):
+                h_kwargs = hlines_kwargs.copy()
+                if backend == "matplotlib" and "linestyle" not in hlines_aes:
+                    h_kwargs.setdefault("linestyle", linestyle_cycle[idx % len(linestyle_cycle)])
+                if "color" not in hlines_aes:
+                    h_kwargs.setdefault("color", _DEFAULT_HLINE_COLOR)
+                if "alpha" not in hlines_aes:
+                    h_kwargs.setdefault("alpha", 0.7)
+
+                h_ds = xr.Dataset({"pareto_k": xr.DataArray(value)})
+                plot_collection.map(
+                    hline,
+                    f"hline_{idx}",
+                    data=h_ds,
+                    ignore_aes="all",
+                    **h_kwargs,
+                )
+
+    if show_bins:
+        bin_text_kwargs = get_visual_kwargs(visuals, "bin_text")
+        if bin_text_kwargs is not False:
+            _, bin_text_aes, _ = filter_aes(plot_collection, aes_by_visuals, "bin_text", [])
+            if "color" not in bin_text_aes:
+                bin_text_kwargs.setdefault("color", "B1")
+            bin_text_kwargs.setdefault("horizontal_align", "center")
+
+            bin_edges = calculate_khat_bin_edges(khat_values, [good_k_threshold, 1.0])
```

Comment:
All this computatiob logic should be in arviz-stats. There were have some functions that are not exposed to user and only serve arviz-plots